src/object.cpp

```diff
@@ -585,15 +626,14 @@ Method *Object::find_method(Env *env, SymbolObject *method_name, MethodVisibilit
         if (visibility >= visibility_at_least) {
             return method;
         } else if (visibility == MethodVisibility::Protected) {
-            env->raise("NoMethodError", "protected method `{}' called for {}", method_name->c_str(), inspect_str(env));
+            GlobalEnv::the()->set_method_missing_reason(MethodMissingReason::Protected);
```

Comment:
This might not be possible because `send(env, "method_missing"_s, ...)` won't allow passing along the `missing_reason`. This is also an issue in CRuby:

```rb
class Foo
  private
  def foo; end
end
f = Foo.new
f.foo                         # => NoMethodError: private method `foo' called for #<Foo:0x00007fd85f832568>
f.send :method_missing, :bar  # => NoMethodError: private method `bar' called for #<Foo:0x00007fd85f832568>
f.bar                         # => NoMethodError: undefined method `bar' for #<Foo:0x00007fd85f832568>
f.send :method_missing, :bar  # => NoMethodError: undefined method `bar' for #<Foo:0x00007fd85f832568>
```

(It's also worth mentioning that the `missing_reason` should be preserved when an override of `method_missing` calls `super`)

Reply:
Fascinating! OK, I see what you mean. Your solution seems to be the best then. 👍